### ISSUE TITLE
Support collaborative editing

### ISSUE THREAD

**Issue by stevage** [tmcw/geojson.net#27]
Sorry if it's too early for the wishlist :) But a need that I've often experienced, and never found a good solution to, is simple collaborative editing of geodata. Basically:

1. I create a dataset.
2. I invite you.
3. You can make some changes.
4. The latest version is always accessible through an API.

uMap kind of supports this, but doesn't have a real GeoJSON API. I'm not sure if this is possible at geojson.io through saving to Github. If it is, it's kind of clunky and not intuitive.

It would be really amazing if this was implemented simply and well.

(I'm not really asking for *simultaneous* collaborative editing here, though at least warning if there has been an intermittent save, and making some effort to merge changes might be nice.)

**Comment by tmcw** [comment]
I'd really want to make this work with the existing GitHub integration; collaborative editing is a huge undertaking and geojson.net (like geojson.io) doesn't have its own backend. Building that backend would be a pretty big task and potentially would require 💸 to keep it running - which as a personal side project, it doesn't have allotted.

That said, my usual policy still holds valid: if someone feels passionately that something should exist and they write a solid implementation with a PR, I'll always happily review and usually merge it. It's more about my own time and resource management that I'm concerned about, so that I can finish projects and keep them from expanding.

**Comment by stevage** [comment]
Yeah, agreed. (I share your aversion to running infrastructure.) I guess the first question is, what's the smoothest workflow that can be achieved using Github? Maybe there are two distinct setups:

1. An existing repo for something like a web app, that contains a couple of geojson files as data. Whoever has commit access to the repo can update the geojson file. (Geojson.net could perhaps add a bit of UX nicety around this, like a link to the repo's contributors management page.)
2. A single file that you want to share with some unique set of contributors to be defined. You'd end up with a single-file repo, which is a bit...weird...but not disastrous. Geojson.net could perhaps handle these situations in a more streamlined situation, rather than forcing you to pick the repo then pick the one file inside it.

It's also possible that there exists some alternative back-end to Github that suits this case. For instance, if myjson.com happened to have an access management feature.

Anyway, the first, simplest step in all the above is to find a better UX for the Github integration that feels more like "collaborative editing" and less like "misusing a version control tool as a storage backend" :) Particularly for people who may barely have heard of Github, let alone have much concept how it works.
